application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -62,4 +69,20 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         }
         message.queue();
     }
+
+    @Override
+    public void onAutoComplete(CommandAutoCompleteInteractionEvent event) {
+        AutoCompleteQuery focusedOption = event.getFocusedOption();
+
+        if (focusedOption.getName().equals(ID_OPTION)) {
+            Collection<Command.Choice> choices = StringDistances
+                .autocompleteSuggestions(focusedOption.getValue(), tagSystem.getAllIds(), 0.5)
+                .stream()
+                .map(id -> new Command.Choice(id, id))
+                .limit(OptionData.MAX_CHOICES)
```

Comment:
this limit should be given to the `autocompleteSuggestions` method instead, it shouldnt sort ALL tags (which could theoretically be thousands). (see my suggested rework)

Reply:
not resolved. u didnt do this at all